### PR TITLE
Remove File's Open Tab When File Deletes

### LINKED ISSUE #381: Remove File's Open Tab When File Deletes
Currently, if a file tab is open and that file is deleted from the left pane, the file still remains open in the tabs.

There are 2 options:
1) When the open file is deleted, the tab with the deleted file should disappear immediately
2) When the open file is deleted, a modal should appear giving a message along the lines of, "this file cannot be deleted because it is currently open."

### PR DESCRIPTION
**Pull request by shivanshs9**
Fixes https://github.com/zowe/zlux/issues/381

Stuffs fixed:
- [x] if a given file is opened and selected, show a snackbar saying that the file can't be deleted.
- [x] if a given file is opened but not selected, close the tab automatically if deleted successfully.

Pre-requisite PRs:
- https://github.com/zowe/zlux-file-explorer/pull/62

### CODE REVIEW

**Comment by shivanshs9** [comment]
@1000TurquoisePogs could you please review it?
As I've mentioned in the description, this PR requires https://github.com/zowe/zlux-file-explorer/pull/62 to be merged into staging and the updated **@zlux/file-explorer** library to be used. That's why Jenkins build is failing.

**Comment by shivanshs9** [comment]
Hey @1000TurquoisePogs @DivergentEuropeans, just wanted to bring attention to this PR. Would you have some time to review it? :smile:

File: webClient/src/app/editor/project-tree/project-tree.component.ts [hunk 1]
```diff
@@ -170,14 +171,30 @@ export class ProjectTreeComponent {
     });
 
     this.editorControl.deleteFile.subscribe(pathAndName => {
-      this.fileExplorer.deleteFileOrFolder(pathAndName);
+      const openedFileContext = this.editorControl.openFileList.getValue().find(fileContext => {
+        return pathAndName === `${fileContext.model.path}/${fileContext.model.fileName}`
```
[the remark this answers is not in the document]
Comment (DivergentEuropeans) [reply webClient/src/app/editor/project-tree/project-tree.component.ts]: Hey @shivanshs9 
tldr; This doesn't work.

Long: This is the right way of doing things, however we don't delete files from the Editor control, we delete files from the right click menu (File Explorer).

The "delete" option this is referring to was when we used to have a "Delete file" menu option in the top-left menubar. However, you had to input the whole path of the file you wanted to delete and it made no sense, so we instead went for the right click option and soon to be, keyboard click, similar to Windows' Explorer. But I apologize, this is on us, when we removed that menu option, we didn't do a good job at deleting or at least commenting some of this code that is no longer used. There are tasks in our backlogs of things to go through and clean up the Editor, but other commitments we have to reach come in the way of that sometimes 🐤 

image

**Review by DivergentEuropeans — CHANGES_REQUESTED** [review]
Hey view my comment, thanks!

File: webClient/src/app/editor/project-tree/project-tree.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shivanshs9) [reply webClient/src/app/editor/project-tree/project-tree.component.ts]: Hey @DivergentEuropeans 
Got it, makes more sense now why some methods are just there with a TODO. Anyways, I've implemented the delete on right click by triggering the delete event from **FileBrowserUSSComponent** all the way up to **ProjectTreeComponent**.
Can you please give some time to test it? Thanks!

**Comment by DivergentEuropeans** [comment]
This has been completed in Editor V2 (v2.x/master)
